### DISCUSSION TITLE
findent.exe not callable on Windows 10.

### DISCUSSION THREAD

**Discussion by himcraft** [krvajal/vscode-fortran-support#527] [Q&A - deprecated see Fortran Discourse] [answered]
When I used the key shortcut `Shift+Alt+F` or clicked 'Format Document' from right-click menu, the extension prompted 'Installing findent.exe through pip with --user option'. After I selected 'install' option, nothing happens, the formatter is still not working.

The output from the terminal is
```
["INFO" - 00:23:51] Formatter located in: C:\\Users\\dude\\AppData\\Roaming\\Python\\bin\\findent.exe
["WARN" - 00:23:51] Formatter: findent.exe not detected in your system.
                                    Attempting to install now.
```

After choose 'install', the output gives
```
["INFO" - 00:23:56] pip install --user --upgrade findent.exe:
--------------------------------------------------------------------------------
--------------------------------------------------------------------------------
["INFO" - 00:23:56] pip install was successful
```

I did install the findent using pip with or without `--user` option, and I changed the path in settings ending with or without `.exe`. But they did not help. I wonder where did I set wrong. Thanks.

---

The relating settings are
```
"fortran.fortls.path": "D:\\\\ProgramFiles\\\\python3_9\\\\Scripts\\\\fortls.exe",
"fortran.formatting.path": "C:\\\\Users\\\\1014h\\\\AppData\\\\Roaming\\\\Python\\\\bin\\\\findent.exe"
```

**Comment by gnikit** [comment, marked as answer]
I suspect your settings paths are wrong. You should probably use a single or double backslash instead of quadruple.

**Reply by himcraft** [reply]
Thanks for your timely response! The quadruple backslash is fine for fortls, as I initially set double backslash in settings and it was automatically translated to four backslashes in the JSON file.

However, after changing the number of backslashes to 2, the extension still could not find the `findent.exe`. Though I can copy the execution file to the source code folder to format manually in the command line as a temp solution.

**Reply by gnikit** [reply]
Oh, my bad, you should just change your `fortran.formatting.path` option to not include the `findent.exe` at the end

**Reply by himcraft** [reply]
Yeah it works! Thank you so much!
